### ISSUE TITLE
Problem when using a custom http.Roundtripper

### ISSUE THREAD

**Issue by lrweck** [steinfletcher/apitest#135]
**Describe the bug**
I've found a weird little bug when using OTEL's RoundTripper.

**Steps to reproduce**
```go
func TestRoundtripper(t *testing.T) {

	newTestEcho := func() *echo.Echo {
		app := echo.New()

		app.GET("/user", func(c echo.Context) error {
			res, err := http.Get("http://localhost:1000/posts")
			if err != nil {
				return c.JSON(400, map[string]any{"err": err.Error()})
			}

			// Also gives an error regardless of a custom HttpClient in apitest
			// cli := &http.Client{
			// 	Transport: otelhttp.NewTransport(http.DefaultTransport),
			// }
			// res, err := cli.Get("http://localhost:1000/posts")
			// if err != nil {
			// 	return c.JSON(400, map[string]any{"err": err.Error()})
			// }

			var mm map[string]any
			json.NewDecoder(res.Body).Decode(&mm)
			return c.JSON(200, mm)
		})

		return app
	}

	m1 := apitest.NewMock().
		Get("http://localhost:1000/posts").RespondWith().Body(`{"result":"ok"}`).Status(http.StatusOK).End().Debug()

	apitest.New().Mocks(m1).Handler(newTestEcho()).
		HttpClient(&http.Client{
			Transport: otelhttp.NewTransport(http.DefaultTransport),
		}).
		Debug().
		Get("/user").
		Expect(t).
		Status(200).End()
}
```
This problem also happens when whatever http.Client calls "http://localhost:1000/posts" is using a custom Roundtripper

**Expected behavior**
not give an error

**Additional context**
When using a custom Roundtripper, apitest ends up not calling RoundTrip itself.

**Comment by lrweck** [comment]
I believe the handling of the transport also causes race conditions when calling with different mocks.

**Comment by lrweck** [comment]
Unfortunately this is not possible. The call http call inside the handler is a simplified view of it. In the scenario I have there are N calls to different endpoints, and each call comes from a different "service". Each may have its own http.Client.

**Comment by stein-f** [comment]
> Unfortunately this is not possible. The call http call inside the handler is a simplified view of it. In the scenario I have there are N calls to different endpoints, and each call comes from a different "service". Each may have its own http.Client.

Yes, I prefer to inject a http client into the handler so the transport can be mocked in tests - it just makes it a lot easier to verify code and run tests quickly. Your other option is to use a real transport, and instead use a localhost URL with a mock server you run as part of the test, but that's not the intention of apitest and the setup is a little more complex.

**Comment by steinfletcher** [comment]
> Unfortunately this is not possible. The call http call inside the handler is a simplified view of it. In the scenario I have there are N calls to different endpoints, and each call comes from a different "service". Each may have its own http.Client.

Yes, if you can't override the http transport of the client that is used internally then you won't be able to use the mocks.

I prefer to make the http client injectable into the handler so that I can mock external calls in tests, rather than the handler instantiating its own client.

Your other option is to run a mock server on localhost and use the existing transport - but that requires some more complex setup that you need to orchestrate from the test.

**Comment by steinfletcher** [comment]
Closing as I don't think this is an issue to solve in aptest
